File: app/src/main/java/moin.movies/receivers/NetworkChangeReceiver.java
package moin.movies.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import moin.movies.views.utils.NetworkUtil;

public class NetworkChangeReceiver extends BroadcastReceiver {

    private static NetworkChange networkChange;

    @Override
    public void onReceive(Context context, Intent intent) {
        networkChange.OnNetworkChanged(NetworkUtil.getConnectivityStatus(context));
    }

    public void setNetworkChange(NetworkChange networkChange) {
        NetworkChangeReceiver.networkChange = networkChange;
    }

    public interface NetworkChange {
        void OnNetworkChanged(int status);
    }

}

File: app/src/main/java/moin.movies/receivers/NetworkChangeReceiver.kt
package moin.movies.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import moin.movies.views.utils.NetworkUtil.getConnectivityStatus

class NetworkChangeReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        networkChange!!.onNetworkChanged(getConnectivityStatus(context))
    }

    fun setNetworkChange(networkChange: NetworkChange?) {
        Companion.networkChange = networkChange
    }

    interface NetworkChange {
        fun onNetworkChanged(status: Int)
    }

    companion object {
        private var networkChange: NetworkChange? = null
    }
}